allennlp/semparse/contexts/table_question_context.py

```diff
@@ -0,0 +1,130 @@
+import re, itertools,nltk, json
+from collections import defaultdict
+from typing import Any, DefaultDict, Dict, List, Tuple, Union, Set
+
+from overrides import overrides
+from unidecode import unidecode
+
+from allennlp.data.tokenizers import Token
+from allennlp.semparse.contexts.knowledge_graph import KnowledgeGraph
+
+# == stop words that will be omitted by ContextGenerator
+STOP_WORDS = ["", "",  "all", "being", "-", "over", "through", "yourselves", "its", "before", 
+             "hadn", "with", "had", ",", "should", "to", "only", "under", "ours", "has", "ought", "do", 
+             "them", "his", "than", "very", "cannot", "they", "not", "during", "yourself", "him", "nor", 
+             "did", "didn", "'ve", "this", "she", "each", "where", "because", "doing", "some", "we", "are", 
+             "further", "ourselves", "out", "what", "for", "weren", "does", "above", "between", "mustn", "?", 
+             "be", "hasn", "who", "were", "here", "shouldn", "let", "hers", "by", "both", "about", "couldn", 
+             "of", "could", "against", "isn", "or", "own", "into", "while", "whom", "down", "wasn", "your", 
+             "from", "her", "their", "aren", "there", "been", ".", "few", "too", "wouldn", "themselves", 
+             ":", "was", "until", "more", "himself", "on", "but", "don", "herself", "haven", "those", "he", 
+             "me", "myself", "these", "up", ";", "below", "'re", "can", "theirs", "my", "and", "would", "then", 
+             "is", "am", "it", "doesn", "an", "as", "itself", "at", "have", "in", "any", "if", "!", "again", "'ll", 
+            "no", "that", "when", "same", "how", "other", "which", "you", "many", "shan", "'t", "'s", "our", "after", 
+            "most", "'d", "such", "'m", "why", "a", "off", "i", "yours", "so", "the", "having", "once"]
+
+                                 
+
+class ContextGenerator:
+    """
+    A Barebones implementation similar to https://github.com/crazydonkey200/neural-symbolic-machines/blob/master/table/wtq/preprocess.py
+    for extracting entities from a question given a table. 
+    """
+
+    def __init__(self, 
+                 cell_values : Set[str], 
+                 column_type_statistics : List[Dict[str,int]],
+                 column_idx_to_name : Dict[int,str]) -> None:
+        self.cell_values = cell_values
+        self.column_types = { column_idx_to_name[column_idx] : max(column_type_statistics[column_idx]) 
+                              for column_idx in column_idx_to_name }
+            
+       
+
+    @classmethod
+    def parse_file(cls, filename: str, question: List[Token], max_tokens_for_num_cell : int) -> 'TableQuestionKnowledgeGraph':
+        f = open(filename, 'r')
```

Comment:
Using a `with open(filename, 'r') as file_pointer:` block instead of this line would automatically close the file when `file_pointer` goes out of scope.

Reply:
Done.